src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -132,6 +132,16 @@ public void copyFrom(MemorySegment src) {
                 base(), min(), size);
     }
 
+    public void copyFromSwap(MemorySegment src, long elemSize) {
```

Comment:
Should this maybe be a public API in MemorySegment as well?

Reply:
> Should this maybe be a public API in MemorySegment as well?

I'll prefer to leave that discussion for another patch

Reply:
Ok. Was just wondering if you had any thoughts about that already :)